pkg/genericapiserver/config.go

```diff
@@ -307,6 +306,10 @@ func (c *Config) ApplyAuthenticationOptions(o *options.BuiltInAuthenticationOpti
 		return c
 	}
 
+	if o.ClientCert != nil && c.SecureServingInfo != nil {
+		c.SecureServingInfo.ClientCA = o.ClientCert.ClientCA
```

Comment:
huh... we now have two potentially (hopefully?) distinct CA pools we'll validate client certs from (to validate request header proxy auth or to verify a client cert for auth directly). We need to include the full pool in the server handshake, so both files need to make it to the secure serving info